src/openms/include/OpenMS/FORMAT/HANDLERS/XMLHandler.h

```diff
@@ -69,27 +72,69 @@ namespace OpenMS
     */
     class OPENMS_DLLAPI StringManager
     {
+
+      typedef std::basic_string<XMLCh> XercesString;
+
+      // Converts from a narrow-character string to a wide-character string.
+      inline XercesString fromNative(const char* str) const
+      {
+        boost::scoped_array<XMLCh> ptr(xercesc::XMLString::transcode(str));
+        return XercesString(ptr.get( ));
```

Comment:
while this is taken from the cookbook, I am not sure whether this is really 100% safe. xerces is quite specific about releasing the memory and by using a `boost::scoped_array<XMLCh>`, we basically do our own memory management here and never tell xerces that we released this string. What if the xerces memory manager tries to do something "smart" like allocating continous memory blocks and we never tell it that  a certain memory area has been released? I think it would be safer to use

```
XMLCh* ptr(xercesc::XMLString::transcode(str));
XercesString result(ptr.get( ));
XMLString::release(ptr);
return result;
```

any thoughts? I am just a bit uncomfortable with this...